lib/semian/protected_resource.rb

```diff
@@ -12,15 +12,19 @@ def initialize(resource, circuit_breaker)
     end
 
     def destroy
-      @resource.destroy
+      @resource.destroy if !!tickets
```

Comment:
No need for double-bang, just do `if tickets`. It'll be `falsy` if there are any. Even if there's `0`, we'll allocate things.